### ISSUE TITLE
NP_BOOLEAN_RETURN_NULL not reported when null is returned via a local variable

### ISSUE THREAD

**Issue by Emilyaxe** [spotbugs/spotbugs#4037]
**Describe the bug**
`NP_BOOLEAN_RETURN_NULL` is reported for `return null;` in a `Boolean` method, but is not triggered when `null` is first assigned to a local variable and then returned. The semantics are identical, leading to a false negative.

**To Reproduce**

```java
public class ReturnNullDemo {

    // BEFORE — NP_BOOLEAN_RETURN_NULL is reported (correct)
    public Boolean isValidDirect(String input) {
        if (input == null) {
            return null;
        }
        return Boolean.TRUE;
    }

    // AFTER — NP_BOOLEAN_RETURN_NULL is NOT reported (false negative)
    public Boolean isValidIndirect(String input) {
        if (input == null) {
            Boolean result = null;
            return result;
        }
        return Boolean.TRUE;
    }

    // AFTER — same issue with a ternary-style method
    public Boolean checkStatus(int code) {
        if (code < 0) {
            Boolean result = null;
            return result;
        }
        return code == 0 ? Boolean.TRUE : Boolean.FALSE;
    }
}
```

Run:

```
spotbugs -textui -effort:max ReturnNullDemo.class
```

**Expected behavior**
`NP_BOOLEAN_RETURN_NULL` should be reported on `isValidIndirect` and `checkStatus` — the methods still return `null` from a `Boolean`-typed method.

**Actual behavior**
No `NP_BOOLEAN_RETURN_NULL` is raised for the indirect variants; only the direct `return null;` form is flagged.

**Environment**

- Java version: JDK 21 (Eclipse Temurin)
- SpotBugs Version: 4.9.8
- How do you use SpotBugs: CLI

**Comment by pradhicsha6** [comment]
I will work on this
